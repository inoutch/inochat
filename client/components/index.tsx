import * as React from "react";

export const Frame: React.SFC = ({children}) => (
    <section className="frame">
        <style jsx>{`
            border: 1px solid grey;
        `}</style>
        {children}
    </section>
);

export const Container: React.SFC = ({children}) => (
    <div className="container">
        <style jsx>{`
            padding-top: 20px;
            margin: 0 auto;
            max-width: 800px;
        `}</style>
        {children}
    </div>
);

export const Divider: React.SFC = () => (
    <div>
        <style jsx>{`
            width: 100%;
            border-bottom: 1px solid grey;
        `}</style>
    </div>
);

interface WindowProps {
    title: string;
}

export const Window: React.SFC<WindowProps> = ({title, children}) => (
    <Frame>
        <h2>
            <style jsx>{`
            padding: 12px;
            `}</style>
            {title}
        </h2>
        <Divider/>
        <div>
            <style jsx>{`
            padding: 12px;
            `}</style>
            {children}
        </div>
    </Frame>
);

export const Header: React.SFC<any> = ({}) => (
    <header>
        <h1>
            <style jsx global>{`
            body {margin: 0;}
            `}</style>
            <style jsx>{`
            padding-left: 12px;
            `}</style>
            INOチャット
        </h1>
        <Divider/>
    </header>
);