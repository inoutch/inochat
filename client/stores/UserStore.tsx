import {action, observable} from "mobx";
import {getCookie, setCookie} from "../utils/cookie";

export class UserStore {
    @observable private token: string = null;
    @observable private username: string = null;
    @observable private userId: number = 0;

    getToken(req?: any) {
        return this.token || getCookie(req).token;
    }

    getUsername(req?: any) {
        return this.username || getCookie(req).username;
    }

    getUserId(req?: any) {
        return this.userId || parseInt(getCookie(req).userId);
    }

    @action setToken(token: string, req?: any) {
        this.token = token;
        setCookie("token", token, req);
    }

    @action setUsername(username: string) {
        this.username = username;
        setCookie("username", username);
    }

    @action setUserId(userId: number) {
        this.userId = userId;
        setCookie("userId", userId);
    }
}