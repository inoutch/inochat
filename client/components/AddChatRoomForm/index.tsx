import * as React from "react";
import Router from "next/router";
import {Window} from "../index";
import {UserStore} from "../../stores/UserStore";
import {post} from "../../utils/net";
import {graphqlEndpoint} from "../../constant/server";
import {inject, observer} from "mobx-react";

interface AddChatRoomProps {
    userStore?: UserStore;
}

interface AddChatRoomState {
    name: string;
}

@inject("userStore")
@observer
export default class AddChatRoomForm extends React.Component<AddChatRoomProps, AddChatRoomState> {
    constructor(props) {
        super(props);
        this.state = {
            name: "",
        };
    }

    async add(event: React.MouseEvent<HTMLElement>) {
        event.preventDefault();
        const query = `
        mutation { 
            newChatRoom(token: \"${this.props.userStore.getToken()}\", name: \"${this.state.name}\") {
                id
            }
        }
        `;
        try {
            const {data} = await post(graphqlEndpoint(), {query});
            Router.push(`/rooms/show?id=${data.newChatRoom.id}`, `/rooms/${data.newChatRoom.id}`);
        } catch (e) {
            alert(e);
        }
    }

    render() {
        return (
            <Window title="部屋を作る">
                <form>
                    <label>
                        ルーム名
                        <input type="text"
                               value={this.state.name}
                               onChange={(e: React.ChangeEvent<any>) => this.setState({name: e.target.value})}/>
                    </label>
                    <input type="submit" onClick={this.add.bind(this)}/>
                </form>
            </Window>
        )
    }
}
