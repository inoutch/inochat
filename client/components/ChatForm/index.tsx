import * as React from "react";
import {joint} from "../../utils/collection";
import {UserStore} from "../../stores/UserStore";

import {Window, Divider} from "../";
import {graphqlEndpoint} from "../../constant/server";
import {post} from "../../utils/net";
import {inject, observer} from "mobx-react";

export interface ChatRoom {
    id: number;
    name: string;
    user: User;
    chats: Chat[];
}

export interface Chat {
    user: User;
    message: string;
}

export interface User {
    id: number;
    username: string;
}

interface ChatFormProps {
    userStore?: UserStore;
    chatRoom: ChatRoom;
    onRefresh: () => void;
}

interface ChatFormState {
    message: string;
    chatRoom: ChatRoom;
}

@inject("userStore")
@observer
export class ChatForm extends React.Component<ChatFormProps, ChatFormState> {
    constructor(props: ChatFormProps) {
        super(props);
        this.state = {
            message: "",
            chatRoom: props.chatRoom,
        };
    }

    async chat(event: React.MouseEvent<HTMLElement>) {
        event.preventDefault();
        const token = this.props.userStore.getToken();
        const message = this.state.message;
        const query = `
            mutation {
                newChat(token: \"${token}\", chatRoomId: ${this.props.chatRoom.id}, message: \"${message}\") {
                    user {
                        username
                    }
                    message
                }
            }
        `;
        try {
            await post(graphqlEndpoint(), {query});
            this.props.onRefresh();
        } catch (e) {
            console.log(e);
            alert("エラー");
        }
    }

    render() {
        return (
            <Window title={this.props.chatRoom.name}>
                <div>チャット({this.props.chatRoom.chats.length})</div>
                <div>
                    <style jsx>{`
                            padding: 12px;
                            `}</style>
                    {joint(this.props.chatRoom.chats.map((chat, i) =>
                        <div key={i*2}>
                            <label>{chat.user.username}</label>
                            <p>{chat.message}</p>
                        </div>
                    ), (_, j) => <Divider key={j}/>)}
                </div>
                <form>
                    <label>チャット内容を入力してください</label>
                    <div>
                        {this.props.userStore.getUsername()}
                    </div>
                    <div>
                        <input type="text"
                               value={this.state.message}
                               onChange={(e) => this.setState({message: e.target.value})}/>
                        <input type="submit" onClick={this.chat.bind(this)}/>
                    </div>
                </form>
            </Window>
        );
    }
}