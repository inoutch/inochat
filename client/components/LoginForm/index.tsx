import * as React from "react";
import Link from "next/link";
import {inject, observer} from "mobx-react";
import {UserStore} from "../../stores/UserStore";
import {makeBasicHeader, get} from "../../utils/net";
import {endpoint} from "../../constant/server";
import {Window} from "../index";
import {redirect} from "../../utils/redirect";

interface LoginFormProps {
    userStore?: UserStore,
}

interface LoginFormState {
    username: string;
    password: string;
}

@inject("userStore")
@observer
export default class RegisterForm extends React.Component<LoginFormProps, LoginFormState> {
    constructor(props: LoginFormProps) {
        super(props);
        this.state = {
            username: "",
            password: "",
        };
    }

    async postUsername(event: React.MouseEvent<HTMLElement>) {
        event.preventDefault();
        try {
            const {token, userId, username} = await get(endpoint() + "/api/login", makeBasicHeader(this.state.username, this.state.password));
            this.props.userStore.setToken(token);
            this.props.userStore.setUsername(username);
            this.props.userStore.setUserId(userId);
            redirect("/rooms");
        } catch (e) {
            alert("エラー");
        }
    }

    render() {
        return (
            <Window title="ログイン">
                <form>
                    <div>
                        <label>ユーザー名</label>
                        <input type="text"
                               value={this.state.username}
                               onChange={(e) => this.setState({username: e.target.value})}/>
                    </div>
                    <div>
                        <label>パスワード</label>
                        <input type="password"
                               value={this.state.password}
                               onChange={(e) => this.setState({password: e.target.value})}/>
                    </div>
                    <div>
                        <Link href="/register">
                            <a>新規登録</a>
                        </Link>
                    </div>
                    <input type="submit" onClick={this.postUsername.bind(this)}/>
                </form>
            </Window>
        );
    }
}