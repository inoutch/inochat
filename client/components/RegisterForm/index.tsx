import * as React from "react";
import {inject, observer} from "mobx-react";
import {UserStore} from "../../stores/UserStore";
import {makeBasicHeader, post, get} from "../../utils/net";
import {endpoint, graphqlEndpoint} from "../../constant/server";
import {Window} from "../index";
import {redirect} from "../../utils/redirect";

interface RegisterFormProps {
    userStore?: UserStore,
}

interface RegisterFormState {
    username: string;
    password: string;
}

@inject("userStore")
@observer
export default class RegisterForm extends React.Component<RegisterFormProps, RegisterFormState> {
    constructor(props: RegisterFormProps) {
        super(props);
        this.state = {
            username: "",
            password: "",
        };
    }

    async postUsername(event: React.MouseEvent<HTMLElement>) {
        event.preventDefault();
        const query = `
        mutation {
            newUser(username: \"${this.state.username}\", password: \"${this.state.password}\") {
                username
            }
        }`;
        try {
            await post(graphqlEndpoint(), {query});
            const {token} = await get(endpoint() + "/api/login", makeBasicHeader(this.state.username, this.state.password));
            this.props.userStore.setToken(token);
            redirect("/rooms");
        } catch (e) {
            alert("エラー");
        }
    }

    render() {
        return (
            <Window title="登録">
                <form>
                    <div>
                        <label>ユーザー名</label>
                        <input type="text"
                               value={this.state.username}
                               onChange={(e) => this.setState({username: e.target.value})}/>
                    </div>
                    <div>
                        <label>パスワード</label>
                        <input type="password"
                               value={this.state.password}
                               onChange={(e) => this.setState({password: e.target.value})}/>
                    </div>
                    <input type="submit" onClick={this.postUsername.bind(this)}/>
                </form>
            </Window>
        );
    }
}