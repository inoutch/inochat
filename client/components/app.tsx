import * as React from "react";
import {Provider} from "mobx-react";
import stores from "../stores/stores";

export class AppProvider extends React.Component {
    render() {
        return (
            <Provider {...stores}>
                <div>
                    {this.props.children}
                </div>
            </Provider>
        )
    }
}