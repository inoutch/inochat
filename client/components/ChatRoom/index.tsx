import * as React from "react";
import Link from "next/link";
import {graphqlEndpoint} from "../../constant/server";
import {post} from "../../utils/net";
import {UserStore} from "../../stores/UserStore";
import {inject, observer} from "mobx-react";
import {User} from "../ChatForm";

interface ChatRoomProps {
    chatRooms: ChatRoom[];
    userStore?: UserStore;
    onUpdateChatRooms: () => void;
}

interface ChatRoom {
    id: number;
    name: string;
    user: User;
}

@inject("userStore")
@observer
export default class extends React.Component<ChatRoomProps> {
    removeChatRooms(token: string, id: number, res?: any) {
        const query = `
        mutation {
            removeChatRoom(token: "${token}", chatRoomId: ${id})
        }
        `;
        return post(graphqlEndpoint(res), {query});
    }

    remove(id: number) {
        this.removeChatRooms(this.props.userStore.getToken(), id)
            .then(this.props.onUpdateChatRooms)
            .catch(() => {});
    }

    render() {
        return (
            <table>
                <thead>
                <tr>
                    <td>部屋名</td>
                    <td>所有者</td>
                    <td>操作</td>
                </tr>
                </thead>
                <tbody>
                {this.props.chatRooms.map((chatRoom, i) =>
                    <tr key={i}>
                        <td>{chatRoom.name}</td>
                        <td>{chatRoom.user.username}</td>
                        <td>
                            <Link as={`/rooms/${chatRoom.id}`} href={`/rooms/show?id=${chatRoom.id}`}>
                                <a className="table-link">閲覧</a>
                            </Link>
                            {chatRoom.user.id === this.props.userStore.getUserId() ?
                                <button onClick={() => this.remove(chatRoom.id)}>削除</button> : null}
                        </td>
                    </tr>)}
                </tbody>
                <style jsx>{`
                table {
                    border-collapse: collapse;
                    margin: 0 auto;
                }
                table, td {
                    border: 1px solid gray;
                }
                td {
                    padding: 18px;
                    min-width: 250px;
                }
                .table-link {
                    padding: 0 8px;
                }
                `}</style>
            </table>
        );
    }
}