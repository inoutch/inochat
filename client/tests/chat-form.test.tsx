import * as React from "react";
import renderer from "react-test-renderer";
import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
enzyme.configure({ adapter: new Adapter() });

import {ChatForm, ChatRoom} from "../components/ChatForm";
import {UserStore} from "../stores/UserStore";
import {Provider} from "mobx-react";

describe("ChatForm Test", () => {
    let chatRoom: ChatRoom;
    let userStore: UserStore;
    beforeEach(() => {
        chatRoom = {
            id: 1,
            name: "部屋名",
            user: {
                id: 1,
                username: "ユーザー名",
            },
            chats: [
                {
                    user: {id: 1, username: "John"},
                    message: "コメント1",
                },
                {
                    user: {id: 2, username: "Steve"},
                    message: "コメント2",
                }
            ]
        };
        userStore = new UserStore();
    });

    test("snapshot", () => {
        const onRefresh = () => {
        };
        const component = renderer.create(
            <Provider userStore={userStore}>
                <ChatForm chatRoom={chatRoom} onRefresh={onRefresh}/>
            </Provider>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test("shallow", () => {
        const onRefresh = () => {
        };
        const component = enzyme.shallow(
            <Provider userStore={userStore}>
                <ChatForm chatRoom={chatRoom} onRefresh={onRefresh}/>
            </Provider>
        );
        expect(component).toMatchSnapshot();
    });
});