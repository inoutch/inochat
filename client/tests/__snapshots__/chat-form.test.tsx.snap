// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatForm Test shallow 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Provider
    userStore={
      UserStore {
        "token": null,
        "userId": 0,
        "username": null,
        Symbol(mobx did run lazy initializers): true,
        Symbol(mobx administration): ObservableObjectAdministration$$1 {
          "defaultEnhancer": [Function],
          "keysAtom": Atom$$1 {
            "diffValue": 0,
            "isBeingObserved": false,
            "isPendingUnobservation": false,
            "lastAccessedBy": 0,
            "lowestObserverState": 2,
            "name": "UserStore@3.keys",
            "observers": Set {},
          },
          "name": "UserStore@3",
          "target": [Circular],
          "values": Map {
            "token" => null,
            "username" => null,
            "userId" => 0,
          },
        },
      }
    }
  >
    <inject-ChatForm-with-userStore
      chatRoom={
        Object {
          "chats": Array [
            Object {
              "message": "コメント1",
              "user": Object {
                "id": 1,
                "username": "John",
              },
            },
            Object {
              "message": "コメント2",
              "user": Object {
                "id": 2,
                "username": "Steve",
              },
            },
          ],
          "id": 1,
          "name": "部屋名",
          "user": Object {
            "id": 1,
            "username": "ユーザー名",
          },
        }
      }
      onRefresh={[Function]}
    />
  </Provider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "chatRoom": Object {
        "chats": Array [
          Object {
            "message": "コメント1",
            "user": Object {
              "id": 1,
              "username": "John",
            },
          },
          Object {
            "message": "コメント2",
            "user": Object {
              "id": 2,
              "username": "Steve",
            },
          },
        ],
        "id": 1,
        "name": "部屋名",
        "user": Object {
          "id": 1,
          "username": "ユーザー名",
        },
      },
      "onRefresh": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "chatRoom": Object {
          "chats": Array [
            Object {
              "message": "コメント1",
              "user": Object {
                "id": 1,
                "username": "John",
              },
            },
            Object {
              "message": "コメント2",
              "user": Object {
                "id": 2,
                "username": "Steve",
              },
            },
          ],
          "id": 1,
          "name": "部屋名",
          "user": Object {
            "id": 1,
            "username": "ユーザー名",
          },
        },
        "onRefresh": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`ChatForm Test snapshot 1`] = `
<section
  className="frame"
>
  <style
    jsx={true}
  >
    
            border: 1px solid grey;
        
  </style>
  <h2>
    <style
      jsx={true}
    >
      
            padding: 12px;
            
    </style>
    部屋名
  </h2>
  <div>
    <style
      jsx={true}
    >
      
            width: 100%;
            border-bottom: 1px solid grey;
        
    </style>
  </div>
  <div>
    <style
      jsx={true}
    >
      
            padding: 12px;
            
    </style>
    <div>
      チャット(
      2
      )
    </div>
    <div>
      <style
        jsx={true}
      >
        
                            padding: 12px;
                            
      </style>
      <div>
        <label>
          John
        </label>
        <p>
          コメント1
        </p>
      </div>
      <div>
        <style
          jsx={true}
        >
          
            width: 100%;
            border-bottom: 1px solid grey;
        
        </style>
      </div>
      <div>
        <label>
          Steve
        </label>
        <p>
          コメント2
        </p>
      </div>
    </div>
    <form>
      <label>
        チャット内容を入力してください
      </label>
      <div />
      <div>
        <input
          onChange={[Function]}
          type="text"
          value=""
        />
        <input
          onClick={[Function]}
          type="submit"
        />
      </div>
    </form>
  </div>
</section>
`;
